src/Mcfedr/QueueManagerBundle/Command/PubSubRunnerCommand.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mcfedr\QueueManagerBundle\Command;
+
+use Google\Cloud\PubSub\Message;
+use Google\Cloud\PubSub\PubSubClient;
+use Mcfedr\QueueManagerBundle\Exception\UnexpectedJobDataException;
+use Mcfedr\QueueManagerBundle\Queue\JobBatch;
+use Mcfedr\QueueManagerBundle\Queue\PubSubJob;
+use Mcfedr\QueueManagerBundle\Runner\JobExecutor;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+class PubSubRunnerCommand extends RunnerCommand
+{
+    /**
+     * @var PubSubClient
+     */
+    private $pubSub;
+
+    /**
+     * @var int
+     */
+    private $batchSize = 10;
+
+    /**
+     * @var int
+     */
+    private $waitTime = 20;
+
+    /**
+     * @var array
+     */
+    private $subscriptions = [];
+
+    /**
+     * @var string
+     */
+    private $defaultSubscription;
+
+    public function __construct(PubSubClient $pubSubClient, string $name, array $options, JobExecutor $jobExecutor, ?LoggerInterface $logger = null)
+    {
+        parent::__construct($name, $options, $jobExecutor, $logger);
+        $this->pubSub = $pubSubClient;
+        $this->setOptions($options);
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+        $this
+            ->addOption('url', null, InputOption::VALUE_REQUIRED, 'The url of SQS queue to run, can be a comma separated list')
+            ->addOption('queue', null, InputOption::VALUE_REQUIRED, 'The name of a queue in the config, can be a comma separated list')
+            ->addOption('batch-size', null, InputOption::VALUE_REQUIRED, 'Number of messages to fetch at once', 10)
+            ->addOption('wait-time', null, InputOption::VALUE_OPTIONAL, 'Wait time between 0-20 seconds')
+        ;
+    }
+
+    protected function getJobs(): ?JobBatch
+    {
+        foreach ($this->subscriptions as $subscription) {
+            $jobs = $this->getJobsFromSubscription($subscription);
+            if ($jobs) {
+                return $jobs;
+            }
+        }
+
+        return null;
+    }
+
+    protected function finishJobs(JobBatch $batch): void
+    {
+        $retryJobs = $batch->getRetries();
+        //todo implement retry failed jobs
+        /** @var PubSubJob[] $toDelete */
+        $toDelete = array_merge($batch->getOks(), $batch->getRetries(), $batch->getFails());
```

Comment:
these are probably called `toAcknowledge` because that's what pub/sub calls this.